Toolkit/Public/Get-RscNasSystem.ps1

```diff
@@ -23,33 +26,57 @@ function Get-RscNasSystem {
 
     .EXAMPLE
     Get details of NAS system with id="72859a28-6276-555a-9a66-d93fe99d2751"
+    Get-RscNasSystem -Id "72859a28-6276-555a-9a66-d93fe99d2751"
+        Or 
     Get-RscNasSystem "72859a28-6276-555a-9a66-d93fe99d2751"
 
     .EXAMPLE
-    Retrieve list of NAS systems.
+    Retrieve list of NAS systems (first 2).
     Get-RscNasSystem -First 2
+    
+    .EXAMPLE
+    Retrieve all NAS systems. 
+    It will return at max 50 items (default page size).
+    Get-RscNasSystem
+
+    .EXAMPLE
+    Retrieve all NAS systems with the name containing "foo".
+    It will return at max 50 items (default page size).
+    Get-RscNasSystem -Name "foo"
+
+    .EXAMPLE
+    Retrieve first 4 NAS systems with the name containing "foo".
+    Get-RscNasSystem -Name "foo" -First 4
     #>
 
     [CmdletBinding(
-        DefaultParameterSetName = "Id"
+        DefaultParameterSetName = "NameAndList"
     )]
     Param(
         # The UUID of the object representing the NAS system.
         [Parameter(
             ParameterSetName = "Id",
-            Mandatory = $true,
+            Mandatory = $false,
             Position = 0
```

Comment:
We should move this to the -Name argument so users can just say `Get-RscNasSystem "foo"`

Reply:
Moved -Name to position 0.